### PR TITLE
ref(migration): Create externaluser table

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
### Objective:
We want to associate Github/Gitlab users and Sentry users. We need to make a new migration to create the `sentry_externaluser` table. 
This is a prerequisite for us to import and convert CODEOWNERS into Sentry IssueOwner syntax.

This PR is first: https://github.com/getsentry/sentry/pull/23696

### CODE REVIEW

File: src/sentry/models/integration.py [hunk 1]
```diff
@@ -62,6 +63,53 @@ class Meta:
...
+    __core__ = False
+
+    team = FlexibleForeignKey("sentry.Team")
+    provider = BoundedPositiveIntegerField(
+        choices=(
+            (ExternalProviders.GITHUB, "github"),
+            (ExternalProviders.GITLAB, "gitlab"),
+        ),
+    )
+    external_id = models.CharField(max_length=64)
```
Comment (mgaeta) [inline src/sentry/models/integration.py:87]: Can you add docstring for this column explaining what it contains?
Also I think 64 might be too low.

File: src/sentry/models/integration.py [hunk 2]
```diff
@@ -62,6 +63,53 @@ class Meta:
...
+    __core__ = False
+
+    user = FlexibleForeignKey("sentry.User")
+    provider = BoundedPositiveIntegerField(
+        choices=(
+            (ExternalProviders.GITHUB, "github"),
+            (ExternalProviders.GITLAB, "gitlab"),
+        ),
+    )
+    external_id = models.CharField(max_length=64)
```
Comment (wedamija) [inline src/sentry/models/integration.py]: `TextField` here too.

File: src/sentry/migrations/0159_create_externaluser_table.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    # You'll also usually want to set this to `False` if you're writing a data
+    # migration, since we don't want the entire migration to run in one long-running
+    # transaction.
+    atomic = True
+
+    dependencies = [
+        ("sentry", "0158_create_externalteam_table"),
+    ]
+
+    operations = [
```
Comment (wedamija) [inline src/sentry/migrations/0159_create_externaluser_table.py:37]: Probably worth just consolidating these into a single migration

File: src/sentry/models/integration.py (hunk 1 above)
Comment (NisanthanNanthakumar) [reply src/sentry/models/integration.py:87]: 👍  bumped it to a textfield

Comment (wedamija) [reply src/sentry/models/integration.py:87]: Need to change this to `TextField` as well as in `ExternalUser`

**Review by wedamija — COMMENTED** [review]
What will these external ids be? Any chance they're emails? Just trying to think about and GDPR considerations

File: src/sentry/migrations/0159_create_externaluser_table.py (hunk 3 above)
Comment (NisanthanNanthakumar) [reply src/sentry/migrations/0159_create_externaluser_table.py:37]: @wedamija is that the best practice? what if there is an issue and we need to rollback a migration. If it's in one, won't it drop both tables?

**Comment by wedamija** [comment]
> What will these external ids be? Any chance they're emails? Just trying to think about and GDPR considerations

^ Just bumping this question

**Comment by NisanthanNanthakumar** [comment]
> > What will these external ids be? Any chance they're emails? Just trying to think about and GDPR considerations
> 
> ^ Just bumping this question

@wedamija I added comments for them in the models. The external_id maps to the Github/Gitlab username & team names.

**Review by wedamija — APPROVED** [review]

**Pull request merged**